Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm

```diff
@@ -1038,7 +1038,41 @@ - (void)_didCommitLayerTree:(const WebKit::RemoteLayerTreeTransaction&)layerTree
             _page->findClient().didAddLayerForFindOverlay(_page.get(), layer);
         }
     }
+
+#if ENABLE(OVERLAY_REGIONS_IN_EVENT_REGION)
+    [self _updateOverlayRegions];
+#endif
+}
+
+#if ENABLE(OVERLAY_REGIONS_IN_EVENT_REGION)
+- (void)_updateOverlayRegions
+{
+    const auto& fixedIDs = _page->scrollingCoordinatorProxy()->scrollingNodeFixedLayerIDs();
+    const auto& remoteLayerTreeHost = downcast<WebKit::RemoteLayerTreeDrawingAreaProxy>(*_page->drawingArea()).remoteLayerTreeHost();
+    NSMutableArray<NSData *> *overlayRegions = [NSMutableArray array];
+    auto offset = [_scrollView contentOffset];
+    for (auto platformID : fixedIDs) {
+        const auto *node = remoteLayerTreeHost.nodeForID(platformID);
+        if (!node)
+            continue;
+
+        CALayer *layer = node->layer();
+        if (!layer)
+            continue;
+
+        auto cgRect = (CGRect)node->eventRegion().region().bounds();
+        if (CGRectIsEmpty(cgRect))
```

Comment:
If the event region is empty, it probably means that the content isn't hit-testable (e.g. `pointer-events none`). Do we want to fall back to layout bounds in this case?

Reply:
I think so. Or some fallback, since otherwise the fixed banner on apple.com does not generate the expected rectangle